### ISSUE TITLE
[Bug Report] Issue with WandB Logging in IsaacLab 2.3.2 and skrl 2.0.0

### ISSUE THREAD

**Issue by BaoyiCui** [isaac-sim/IsaacLab#5252]
### Describe the bug

I'm currently using IsaacLab version 2.3.2 and SKRL version 2.0.0. I installed the packages following the official documentation ("installation using Isaac Sim pip package"). During installation, I noticed that SKRL 2.0.0 was installed (previous versions used SKRL 1.4.3). I then created an external Direct Multi-Agent environment with RL library set to SKRL and MAPPO algorithm.

Here’s the configuration section of my training setup where I attempted to enable WandB logging:
```yaml
...
# logging and checkpoint
  experiment:
    directory: "cart_double_pendulum_direct"
    experiment_name: ""
    write_interval: auto
    checkpoint_interval: auto
    wandb: True    # newly added
```
Hoever, when I started the training, I encountered the following warnings:
```txt
[skrl:WARNING] The 'lambda' field in the configuration is deprecated. Use 'gae_lambda' instead
19:20:35 [runner.py] WARNING: The 'lambda' field in the configuration is deprecated. Use 'gae_lambda' instead
[skrl:WARNING] The 'clip_predicted_values' field in the configuration is deprecated. Use a 'value_clip' value greater than 0 to clip the predicted values instead
19:20:35 [runner.py] WARNING: The 'clip_predicted_values' field in the configuration is deprecated. Use a 'value_clip' value greater than 0 to clip the predicted values instead
[skrl:WARNING] The 'state_preprocessor' field in the configuration has been replaced by 'observation_preprocessor'. If the 'state_preprocessor' definition is desired but the `observation_preprocessor` is not, define the last one to None (null) to avoid the automatic replacement
19:20:35 [runner.py] WARNING: The 'state_preprocessor' field in the configuration has been replaced by 'observation_preprocessor'. If the 'state_preprocessor' definition is desired but the `observation_preprocessor` is not, define the last one to None (null) to avoid the automatic replacement
[skrl:WARNING] The 'shared_state_preprocessor' field in the configuration is deprecated. Use 'state_preprocessor' instead
19:20:35 [runner.py] WARNING: The 'shared_state_preprocessor' field in the configuration is deprecated. Use 'state_preprocessor' instead
```

Following these warnings, i checked WandB logs, but unlike previous version of skrl 1.4.3, where metrics like reward, loss, etc., were correctly logged, in this case, only system-related metrics (e.g., system time, memory usage) are being recorded.

<img width="1432" height="1021" alt="Image" src="https://github.com/user-attachments/assets/c165e70b-dec6-4611-916d-0e4699344992" />

### Steps to reproduce

1. Installed IsaacLab version  and SKRL version 2.0.0 using the command pip install isaaclab.
2. Created an external Direct Multi-Agent environment.
3. Set the RL library to SKRL and algorithm to MAPPO.
4. Enabled WandB logging in the configuration file by setting `wandb: True`.
5. Observed that only system-related metrics were being recorded, and other metrics like loss and reward were missing.

### System Info

Describe the characteristic of your environment:

- Isaac Sim Version: 5.1
- Isaac Lab Version: 2.3.2
- skrl Version: 2.0.0
- OS: Ubuntu 24.04
- GPU: RTX 5090
- CUDA: 13.1
- GPU Driver: 590.48.01

### Additional context
I have resolved the logging issue by modifying the train.py file as shown below. I hope this helps you address the bug.
```python
# Copyright (c) 2022-2026, The Isaac Lab Project Developers (https://github.com/isaac-sim/IsaacLab/blob/main/CONTRIBUTORS.md).
# All rights reserved.
#
# SPDX-License-Identifier: BSD-3-Clause

"""
Script to train RL agent with skrl.

Visit the skrl documentation (https://skrl.readthedocs.io) to see the examples structured in
a more user-friendly way.
"""

"""Launch Isaac Sim Simulator first."""

import argparse
import sys

from isaaclab.app import AppLauncher

# add argparse arguments
parser = argparse.ArgumentParser(description="Train an RL agent with skrl.")
parser.add_argument("--video", action="store_true", default=False, help="Record videos during training.")
parser.add_argument("--video_length", type=int, default=200, help="Length of the recorded video (in steps).")
parser.add_argument("--video_interval", type=int, default=2000, help="Interval between video recordings (in steps).")
parser.add_argument("--num_envs", type=int, default=None, help="Number of environments to simulate.")
parser.add_argument("--task", type=str, default=None, help="Name of the task.")
parser.add_argument(
    "--agent",
    type=str,
    default=None,
    help=(
        "Name of the RL agent configuration entry point. Defaults to None, in which case the argument "
        "--algorithm is used to determine the default agent configuration entry point."
    ),
)
parser.add_argument("--seed", type=int, default=None, help="Seed used for the environment")
parser.add_argument(
    "--distributed", action="store_true", default=False, help="Run training with multiple GPUs or nodes."
)
parser.add_argument("--checkpoint", type=str, default=None, help="Path to model checkpoint to resume training.")
parser.add_argument("--max_iterations", type=int, default=None, help="RL Policy training iterations.")
parser.add_argument("--export_io_descriptors", action="store_true", default=False, help="Export IO descriptors.")
parser.add_argument(
    "--ml_framework",
    type=str,
    default="torch",
    choices=["torch", "jax", "jax-numpy"],
    help="The ML framework used for training the skrl agent.",
)
parser.add_argument(
    "--algorithm",
    type=str,
    default="PPO",
    choices=["AMP", "PPO", "IPPO", "MAPPO"],
    help="The RL algorithm used for training the skrl agent.",
)
parser.add_argument(
    "--ray-proc-id", "-rid", type=int, default=None, help="Automatically configured by Ray integration, otherwise None."
)
# append AppLauncher cli args
AppLauncher.add_app_launcher_args(parser)
# parse the arguments
args_cli, hydra_args = parser.parse_known_args()
# always enable cameras to record video
if args_cli.video:
    args_cli.enable_cameras = True

# clear out sys.argv for Hydra
sys.argv = [sys.argv[0]] + hydra_args

# launch omniverse app
app_launcher = AppLauncher(args_cli)
simulation_app = app_launcher.app

"""Rest everything follows."""

import logging
import os
import random
import time
from datetime import datetime

import numpy as np
import gymnasium as gym
import skrl
from packaging import version

# check for minimum supported skrl version
SKRL_VERSION = "1.4.3"
if version.parse(skrl.__version__) < version.parse(SKRL_VERSION):
    skrl.logger.error(
        f"Unsupported skrl version: {skrl.__version__}. "
        f"Install supported version using 'pip install skrl>={SKRL_VERSION}'"
    )
    exit()

if args_cli.ml_framework.startswith("torch"):
    from skrl.utils.runner.torch import Runner
elif args_cli.ml_framework.startswith("jax"):
    from skrl.utils.runner.jax import Runner

from isaaclab.envs import (
    DirectMARLEnv,
    DirectMARLEnvCfg,
    DirectRLEnvCfg,
    ManagerBasedRLEnvCfg,
    multi_agent_to_single_agent,
)
from isaaclab.utils.assets import retrieve_file_path
from isaaclab.utils.dict import print_dict
from isaaclab.utils.io import dump_yaml

from isaaclab_rl.skrl import SkrlVecEnvWrapper

import isaaclab_tasks  # noqa: F401
from isaaclab_tasks.utils.hydra import hydra_task_config

# import logger
logger = logging.getLogger(__name__)

import mape.tasks  # noqa: F401

# ---------------------------------------------------------------------------
# Fix: skrl uses its own SummaryWriter (skrl.utils.tensorboard.SummaryWriter)
# which writes TensorBoard events directly via low-level APIs, bypassing the
# PyTorch / TF writers that wandb's sync_tensorboard=True monkey-patches.
# As a result, wandb never receives any training metrics even though the
# TensorBoard event files are written correctly.
#
# Solution: patch MultiAgent.write_tracking_data so that every time skrl
# would flush metrics to TensorBoard it also calls wandb.log() directly.
#
# X-axis fix: skrl sets sync_tensorboard=True by default, which causes wandb
# to ignore the explicit step= argument and use its own internal commit
# counter (0, 1, 2 ...) as the x-axis. We patch init() so sync_tensorboard
# is forced to False, which lets our explicit step values appear correctly.
# ---------------------------------------------------------------------------
import wandb as _wandb_module

_orig_wandb_init = _wandb_module.init


def _patched_wandb_init(*args, **kwargs):
    kwargs["sync_tensorboard"] = False
    return _orig_wandb_init(*args, **kwargs)


_wandb_module.init = _patched_wandb_init

from skrl.multi_agents.torch.base import MultiAgent as _SkrlMultiAgent

_orig_write_tracking_data = _SkrlMultiAgent.write_tracking_data


def _patched_write_tracking_data(self, *, timestep: int, timesteps: int) -> None:
    # Collect aggregated metrics BEFORE the original method clears tracking_data.
    wandb_data: dict = {}
    for k, v in self.tracking_data.items():
        if k.endswith("(min)"):
            wandb_data[k] = float(np.min(v))
        elif k.endswith("(max)"):
            wandb_data[k] = float(np.max(v))
        else:
            wandb_data[k] = float(np.mean(v))

    # Original behaviour: write to TensorBoard and clear tracking_data.
    _orig_write_tracking_data(self, timestep=timestep, timesteps=timesteps)

    # Additionally push metrics to wandb if a run is active.
    if _wandb_module.run is not None and wandb_data:
        _wandb_module.log(wandb_data, step=timestep)


_SkrlMultiAgent.write_tracking_data = _patched_write_tracking_data
# ---------------------------------------------------------------------------

# config shortcuts
if args_cli.agent is None:
    algorithm = args_cli.algorithm.lower()
    agent_cfg_entry_point = "skrl_cfg_entry_point" if algorithm in ["ppo"] else f"skrl_{algorithm}_cfg_entry_point"
else:
    agent_cfg_entry_point = args_cli.agent
    algorithm = agent_cfg_entry_point.split("_cfg")[0].split("skrl_")[-1].lower()


@hydra_task_config(args_cli.task, agent_cfg_entry_point)
def main(env_cfg: ManagerBasedRLEnvCfg | DirectRLEnvCfg | DirectMARLEnvCfg, agent_cfg: dict):
    """Train with skrl agent."""
    # override configurations with non-hydra CLI arguments
    env_cfg.scene.num_envs = args_cli.num_envs if args_cli.num_envs is not None else env_cfg.scene.num_envs
    env_cfg.sim.device = args_cli.device if args_cli.device is not None else env_cfg.sim.device

    # check for invalid combination of CPU device with distributed training
    if args_cli.distributed and args_cli.device is not None and "cpu" in args_cli.device:
        raise ValueError(
            "Distributed training is not supported when using CPU device. "
            "Please use GPU device (e.g., --device cuda) for distributed training."
        )

    # multi-gpu training config
    if args_cli.distributed:
        env_cfg.sim.device = f"cuda:{app_launcher.local_rank}"
    # max iterations for training
    if args_cli.max_iterations:
        agent_cfg["trainer"]["timesteps"] = args_cli.max_iterations * agent_cfg["agent"]["rollouts"]
    agent_cfg["trainer"]["close_environment_at_exit"] = False
    # configure the ML framework into the global skrl variable
    if args_cli.ml_framework.startswith("jax"):
        skrl.config.jax.backend = "jax" if args_cli.ml_framework == "jax" else "numpy"

    # randomly sample a seed if seed = -1
    if args_cli.seed == -1:
        args_cli.seed = random.randint(0, 10000)

    # set the agent and environment seed from command line
    # note: certain randomization occur in the environment initialization so we set the seed here
    agent_cfg["seed"] = args_cli.seed if args_cli.seed is not None else agent_cfg["seed"]
    env_cfg.seed = agent_cfg["seed"]

    # specify directory for logging experiments
    log_root_path = os.path.join("logs", "skrl", agent_cfg["agent"]["experiment"]["directory"])
    log_root_path = os.path.abspath(log_root_path)
    print(f"[INFO] Logging experiment in directory: {log_root_path}")
    # specify directory for logging runs: {time-stamp}_{run_name}
    log_dir = datetime.now().strftime("%Y-%m-%d_%H-%M-%S") + f"_{algorithm}_{args_cli.ml_framework}"
    # The Ray Tune workflow extracts experiment name using the logging line below, hence,
    # do not change it (see PR #2346, comment-2819298849)
    print(f"Exact experiment name requested from command line: {log_dir}")
    if agent_cfg["agent"]["experiment"]["experiment_name"]:
        log_dir += f"_{agent_cfg['agent']['experiment']['experiment_name']}"
    # set directory into agent config
    agent_cfg["agent"]["experiment"]["directory"] = log_root_path
    agent_cfg["agent"]["experiment"]["experiment_name"] = log_dir
    # update log_dir
    log_dir = os.path.join(log_root_path, log_dir)

    # dump the configuration into log-directory
    dump_yaml(os.path.join(log_dir, "params", "env.yaml"), env_cfg)
    dump_yaml(os.path.join(log_dir, "params", "agent.yaml"), agent_cfg)

    # get checkpoint path (to resume training)
    resume_path = retrieve_file_path(args_cli.checkpoint) if args_cli.checkpoint else None

    # set the IO descriptors export flag if requested
    if isinstance(env_cfg, ManagerBasedRLEnvCfg):
        env_cfg.export_io_descriptors = args_cli.export_io_descriptors
    else:
        logger.warning(
            "IO descriptors are only supported for manager based RL environments. No IO descriptors will be exported."
        )

    # set the log directory for the environment (works for all environment types)
    env_cfg.log_dir = log_dir

    # create isaac environment
    env = gym.make(args_cli.task, cfg=env_cfg, render_mode="rgb_array" if args_cli.video else None)

    # convert to single-agent instance if required by the RL algorithm
    if isinstance(env.unwrapped, DirectMARLEnv) and algorithm in ["ppo"]:
        env = multi_agent_to_single_agent(env)

    # wrap for video recording
    if args_cli.video:
        video_kwargs = {
            "video_folder": os.path.join(log_dir, "videos", "train"),
            "step_trigger": lambda step: step % args_cli.video_interval == 0,
            "video_length": args_cli.video_length,
            "disable_logger": True,
        }
        print("[INFO] Recording videos during training.")
        print_dict(video_kwargs, nesting=4)
        env = gym.wrappers.RecordVideo(env, **video_kwargs)

    start_time = time.time()

    # wrap around environment for skrl
    env = SkrlVecEnvWrapper(env, ml_framework=args_cli.ml_framework)  # same as: `wrap_env(env, wrapper="auto")`

    # configure and instantiate the skrl runner
    # https://skrl.readthedocs.io/en/latest/api/utils/runner.html
    runner = Runner(env, agent_cfg)

    # load checkpoint (if specified)
    if resume_path:
        print(f"[INFO] Loading model checkpoint from: {resume_path}")
        runner.agent.load(resume_path)

    # run training
    runner.run()

    print(f"Training time: {round(time.time() - start_time, 2)} seconds")

    # close the simulator
    env.close()


if __name__ == "__main__":
    # run the main function
    main()
    # close sim app
    simulation_app.close()

```

### Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have checked that the issue is not in running Isaac Sim itself and is related to the repo

### Acceptance Criteria

Add the criteria for which this task is considered **done**. If not known at issue creation time, you can add this once the issue is assigned.

- [ ] No warnings about deprecated parameters when using SKRL 2.0.0.
- [ ] WandB logging works correctly and records relevant metrics, similar to SKRL 1.4.3.

**Comment by RandomOakForest** [comment]
Thank you for posting this. The team will review it.

**Comment by PeterL-NV** [comment]
Hi, Thanks for reporting your issue! Here's what's broken:

  1. skrl 2.0+ ships its own `SummaryWriter` at `skrl/utils/tensorboard.py` that writes TensorBoard event protobufs directly through `tensorboard.summary.writer.event_file_writer.EventFileWriter` — bypassing `torch.utils.tensorboard.SummaryWriter` (and the TF equivalent). wandb's `sync_tensorboard=True` mode hooks into those PyTorch/TF SummaryWriter classes. Since skrl never instantiates one, neither the tfevent-file registration nor the add_scalar interception ever fires, and wandb has nothing to forward besides system telemetry.
  2. skrl's `Agent.write_tracking_data` (and the multi-agent equivalent) only call `self.writer.add_scalar(...)` — it never invokes wandb.log(). So even with sync_tensorboard disabled, no training data reaches wandb on its own.
  3. `sync_tensorboard=True` mutates `wandb.log()` to ignore explicit `step=` arguments (wandb uses its own internal commit counter). So if you just add wandb.log(...) without flipping sync_tensorboard off first, the x-axis is wrong. `skrl/agents/torch/base.py` sets  `wandb_kwargs.setdefault("sync_tensorboard", True)`, so you get the broken combination by default whenever `experiment.wandb: true` is set.

Strictly speaking this is an upstream skrl 2.0.0 bug, not an IsaacLab bug, I would suggest you keep your current WAR until this is fixed in upstream skrl.

**Comment by PeterL-NV** [comment]
Closing as this is an upstream bug.
